modules/mono/csharp_script.cpp

```diff
@@ -2207,11 +2250,16 @@ void CSharpScript::reload_registered_script(Ref<CSharpScript> p_script) {
 	update_script_class_info(p_script);
 
 	p_script->_update_exports();
+
+#if TOOLS_ENABLED
+	EditorFileSystem::get_singleton()->update_file(p_script->get_path());
```

Comment:
```suggestion
	// If the EditorFileSystem singleton is available, update the file;
	// otherwise, the file will be updated when the singleton becomes available.
	EditorFileSystem *efs = EditorFileSystem::get_singleton();
	if (efs) {
		efs->update_file(p_script->get_path());
	}
```

efs can be null here as well during resource loader initialization.

Not sure the one in reload can also hit this issue, I didn't crash there during my testing.

(Not sure what I am doing to hit all of these)

Reply:
I thought I had tested well enough, but you are right I missed this. I changed the one in `reload` as well, just to be safe.